Code/Engine/RendererCore/Debug/Implementation/DebugRenderer.cpp

```diff
@@ -1252,6 +1281,58 @@ void ezDebugRenderer::RenderInternal(const ezDebugRendererContext& context, cons
     }
   }
 
+  // info text
+  {
+    static_assert((int)ezDebugRenderer::ScreenPlacement::ENUM_COUNT == 6);
+
+    HorizontalAlignment ha[(int)ezDebugRenderer::ScreenPlacement::ENUM_COUNT] = {
+      HorizontalAlignment::Left,
+      HorizontalAlignment::Center,
+      HorizontalAlignment::Right,
+      HorizontalAlignment::Left,
+      HorizontalAlignment::Center,
+      HorizontalAlignment::Right};
+
+    VerticalAlignment va[(int)ezDebugRenderer::ScreenPlacement::ENUM_COUNT] = {
+      VerticalAlignment::Top,
+      VerticalAlignment::Top,
+      VerticalAlignment::Top,
+      VerticalAlignment::Bottom,
+      VerticalAlignment::Bottom,
+      VerticalAlignment::Bottom};
+
+    int offs[(int)ezDebugRenderer::ScreenPlacement::ENUM_COUNT] = {20, 20, 20, -20, -20, -20};
+
+    ezInt32 resX = (ezInt32)renderViewContext.m_pViewData->m_ViewPortRect.width;
+    ezInt32 resY = (ezInt32)renderViewContext.m_pViewData->m_ViewPortRect.height;
+
+    ezVec2I32 anchor[(int)ezDebugRenderer::ScreenPlacement::ENUM_COUNT] = {
+      ezVec2I32(10, 10),
+      ezVec2I32(resX / 2, 10),
+      ezVec2I32(resX - 10, 10),
+      ezVec2I32(10, resY - 10),
+      ezVec2I32(resX / 2, resY - 10),
+      ezVec2I32(resX - 10, resY - 10)};
+
+    for (ezUInt32 corner = 0; corner < (ezUInt32)ezDebugRenderer::ScreenPlacement::ENUM_COUNT; ++corner)
+    {
+      auto& cd = pData->m_infoTextData[corner];
+
+      cd.Sort([](const InfoTextData& lhs, const InfoTextData& rhs) -> bool { return lhs.m_group < rhs.m_group; });
```

Comment:
Sort is not stable so text lines might be shuffled within one group

Reply:
actually the sort is not needed at all

Reply:
The sort is needed, otherwise there is no deterministic ordering. All debug render function can be called from multiple threads.

Reply:
Ah, right. I've switched to insertion sort. Maybe we should have a "StableSort" function on our containers ?

Reply:
Unfortunately there is no fast and stable sort algorithm that works entirely in-place. Insertion sort is stable but also slow^^

Reply:
Yeah, I meant to add "StableSort" as a function to the containers, so that you can pick Sort or StableSort, depending on your needs. Right now it is not obvious that Sort isn't stable, and what to do, if you need it.

Reply:
Ok, doesn't hurt to add a StableSort function to the containers, but we should document that it is not really fast.